pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/security/AuthenticationProvider.java

```diff
@@ -39,7 +40,9 @@
 
 	private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationProvider.class);
 
-	private RestTemplate restTemplate = new RestTemplate();
+	@Autowired
+	@Qualifier(value = "restTemplateConfig")
```

Comment:
Why do you need the qualifier for this bean and not in auth token util class ?

Reply:
Since there were two RestTemplate in pre-reg Application service, i had to use Qualifier. Another restTemplate is created for Test code. AuthtokenUtil comes from Pre-reg Core and only single RestTemplate is created there.